### ISSUE TITLE
Bug: Name resolution for No-Ip ddnskeys

### ISSUE THREAD

**Issue by Knokey** [qdm12/ddns-updater#712]
**TLDR**: Name resolution for No-Ip ddnskeys

1. Is this urgent: No
2. DNS provider(s) you use: No-Ip
3. Program version: 2024-05-06T11:16:06.246Z (commit 542e895)
4. What are you using to run the container: ansible playbook
5. Extra information

Logs:

```log
2024-05-07T07:26:31Z INFO Settings summary:
├── HTTP client
|   └── Timeout: 10s
├── Update
|   ├── Period: 5m0s
|   └── Cooldown: 5m0s
├── Public IP fetching
|   ├── HTTP enabled: yes
|   ├── HTTP IP providers
|   |   └── all
|   ├── HTTP IPv4 providers
|   |   └── all
|   ├── HTTP IPv6 providers
|   |   └── all
|   ├── DNS enabled: yes
|   ├── DNS timeout: 3s
|   └── DNS over TLS providers
|       └── all
├── Resolver: use Go default resolver
├── Server
|   ├── Listening address: :8000
|   └── Root URL: /
├── Health
|   ├── Server listening address: 127.0.0.1:9999
|   ├── Healthchecks.io base URL: https://xxxx/ping
|   └── Healthchecks.io UUID: xxxx
├── Paths
|   └── Data directory: /updater/data
├── Backup: disabled
└── Logger
    ├── Level: info
    └── Caller: hidden
2024-05-07T07:26:31Z INFO reading JSON config from file /updater/data/config.json
2024-05-07T07:26:31Z INFO Found 2 settings to update records
2024-05-07T07:26:31Z INFO Reading history from database: domain all.ddnskey.com host @ ipv4
2024-05-07T07:26:31Z INFO Reading history from database: domain xxxx host xxxx ipv4
2024-05-07T07:26:31Z INFO [backup] disabled
2024-05-07T07:26:31Z INFO [healthcheck server] listening on 127.0.0.1:9999
2024-05-07T07:26:31Z INFO [http server] listening on :8000
2024-05-07T07:26:31Z INFO obtaining ipv4 address succeeded after 1 failed try
2024/05/07 07:26:34 "GET http://xxxx/ HTTP/1.0" from 172.17.0.1:52862 - 200 2147B in 334.673µs
2024-05-07T07:31:31Z INFO ipv4 address of all.ddnskey.com is 34.198.182.201 and your ipv4 address  is 93.239.x.x
2024-05-07T07:31:31Z INFO Updating record [domain: all.ddnskey.com | host: @ | provider: noip | ip: ipv4] to use 93.239.x.x
2024-05-07T07:41:31Z INFO ipv4 address of all.ddnskey.com is 34.198.182.201 and your ipv4 address  is 93.239.x.x
2024-05-07T07:41:31Z INFO Updating record [domain: all.ddnskey.com | host: @ | provider: noip | ip: ipv4] to use 93.239.x.x
```

Configuration file

```json
{
    "settings": [
        {
            "provider": "noip",
            "domain": "all.ddnskey.com",
            "host": "@",
            "username": "xxxx",
            "password": "xxxx",
            "ip_version": "ipv4",
            "ipv6_suffix": "",
            "provide_ip": false
        },
        {
            "provider": "cloudflare",
            "zone_identifier": "xxxx",
            "domain": "xxxx",
            "host": "xxxx",
            "ttl": 600,
            "token": "xxxx",
            "ip_version": "ipv4",
            "ipv6_suffix": ""
        }
    ]
}
```

Host OS: Ubuntu 22.04.4

I encountered the following problem: Updating the ip for No-Ip with ddnskey works, but ddns updater cannot verify the updated ip. The reason is that it tries to match the updated ip against the ip of api domain of no-ip. With a ddnskey from no-ip, all domains need to access the same url (all.ddnskey.com) and no-ip is matching the domain to upgrade via the ddnskey (username). 

Solution would be to an extra field for no-ip where the actual domain that will be updated can be defined. Right now, dens updater does not know for which domain the ip address has been updated.

Output of the health check:

```log
ERROR program is unhealthy: lookup IP addresses do not match: 34.198.182.201 instead of 93.239.x.x for all.ddnskey.com
```

Docs for ddnskey at no-ip: https://www.noip.com/support/knowledgebase/how-to-setup-and-use-a-ddns-key

**Comment by qdm12** [comment]
Thanks for creating the issue 👍 

> A solution would be to an extra field for no-ip where the actual domain that will be updated can be defined.

In that case, wouldn't it be simpler to just update each subdomain in a separate setting block, instead of using `all.<your-ddnskey>.com`? Or is this not possible 🤔 ?

**Comment by Knokey** [comment]
Thanks for the response.

No, I think this is not possible. `all.ddnskey.com` includes no placeholder. Setting this as your domain is mandatory that no-ip can determine you want to update via a ddnskey.

Excerpt from the no-ip documentation:

```
The DDNS settings may look different depending on what device is being configured, but the following information will be requested:
DDNS Provider: No-IP
Username: Your DDNS Key Username / Email
Password: Your DDNS Key Password
Hostname/Domain: all.ddnskey.com
Server/Server Address: dynupdate.no-ip.com
Not every device will ask you for a server or server address. The service will still work without issue, if the device does not ask for it.
```

**Comment by Aypeus** [comment]
Is issue related to why Docker gives the status of this being `(unhealthy)`? _I'm asking so I don't go ahead and create a issue if the problem is the same_
I attached my `docker logs ..`
[ddns-updater-log.txt](https://github.com/qdm12/ddns-updater/files/15256155/ddns-updater-log.txt)
 output as a .txt in my post.

**Comment by Knokey** [comment]
I think your health check issue is different from mine. I get a wrong ip reported, because ddns updater cannot know the hostname that is updated and your health check is timed out.

**Comment by Aypeus** [comment]
> I think your health check issue is different from mine. I get a wrong ip reported, because ddns updater cannot know the hostname that is updated and your health check is timed out.

I see, thank you for pointing that out! _so I should go ahead and post my problem as a new issue then i suppose_ Good luck to!

**Comment by king8084** [comment]
sad to see this isnt implemented :(
